OpenRA.Game/GameRules/RulesetCache.cs

```diff
@@ -60,38 +62,48 @@ public Ruleset Load(IReadOnlyFileSystem fileSystem, Map map = null)
 			Dictionary<string, TileSet> tileSets;
 
 			using (new PerfTimer("Actors"))
-				actors = LoadYamlRules(fileSystem, actorCache, m.Rules,
-					map != null ? map.RuleDefinitions : NoMapRules,
+				actors = LoadYamlRules(fileSystem, actorCache, m.Rules, additionalRules,
 					k => new ActorInfo(Game.ModData.ObjectCreator, k.Key.ToLowerInvariant(), k.Value));
 
 			using (new PerfTimer("Weapons"))
-				weapons = LoadYamlRules(fileSystem, weaponCache, m.Weapons,
-					map != null ? map.WeaponDefinitions : NoMapRules,
+				weapons = LoadYamlRules(fileSystem, weaponCache, m.Weapons, additionalWeapons,
 					k => new WeaponInfo(k.Key.ToLowerInvariant(), k.Value));
 
 			using (new PerfTimer("Voices"))
-				voices = LoadYamlRules(fileSystem, voiceCache, m.Voices,
-					map != null ? map.VoiceDefinitions : NoMapRules,
+				voices = LoadYamlRules(fileSystem, voiceCache, m.Voices, additionalVoices,
 					k => new SoundInfo(k.Value));
 
 			using (new PerfTimer("Notifications"))
-				notifications = LoadYamlRules(fileSystem, notificationCache, m.Notifications,
-					map != null ? map.NotificationDefinitions : NoMapRules,
+				notifications = LoadYamlRules(fileSystem, notificationCache, m.Notifications, additionalNotifications,
 					k => new SoundInfo(k.Value));
 
 			using (new PerfTimer("Music"))
-				music = LoadYamlRules(fileSystem, musicCache, m.Music,
-					map != null ? map.MusicDefinitions : NoMapRules,
+				music = LoadYamlRules(fileSystem, musicCache, m.Music, additionalMusic,
 					k => new MusicInfo(k.Key, k.Value));
 
 			using (new PerfTimer("TileSets"))
 				tileSets = LoadTileSets(fileSystem, tileSetCache, m.TileSets);
 
 			// TODO: only initialize, and then cache, the provider for the given map
-			var sequences = tileSets.ToDictionary(t => t.Key, t => new SequenceProvider(fileSystem, modData, t.Value, map));
+			var sequences = tileSets.ToDictionary(t => t.Key, t => new SequenceProvider(fileSystem, modData, t.Value, additionalSequences));
 			return new Ruleset(actors, weapons, voices, notifications, music, tileSets, sequences);
 		}
 
+		/// <summary>
+		/// Cache and return the Ruleset for a given map.
+		/// If a map isn't specified then return the default mod Ruleset.
+		/// </summary>
+		public Ruleset Load(IReadOnlyFileSystem fileSystem, Map map = null)
+		{
+			return Load(fileSystem,
+				map != null ? map.RuleDefinitions : NoMapRules,
```

Comment:
Couldn't we do something like
`return map == null ? Load(fileSystem, NoMapRules, NoMapRules,...) : Load(fileSystem, map.RuleDefinitions, map.WeaponDefinitions, ...);` here?